bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v2/PWAImplTest.java

```diff
@@ -0,0 +1,135 @@
+
+import org.apache.sling.api.resource.ModifiableValueMap;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import com.adobe.cq.wcm.core.components.context.CoreComponentTestContext;
+import com.adobe.cq.wcm.core.components.models.PWA;
+import com.day.cq.commons.jcr.JcrConstants;
+import io.wcm.testing.mock.aem.junit5.AemContext;
+import io.wcm.testing.mock.aem.junit5.AemContextExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(AemContextExtension.class)
+public class PWAImplTest {
+
+    private static final String SITES_PAGE_PATH = "/content/mysite/en";
+    private ResourceResolver resolver;
+    private PWA pwa;
+    private Resource resource;
+    private ModifiableValueMap mvp;
+
+    protected final AemContext context = CoreComponentTestContext.newAemContext();
+
+    @BeforeEach
+    public void setUp() {
+        context.load().json("/pwa" + CoreComponentTestContext.TEST_CONTENT_JSON, SITES_PAGE_PATH);
+        resolver = context.resourceResolver();
+        resource = spy(resolver.getResource(SITES_PAGE_PATH));
+        mvp = resource.adaptTo(ModifiableValueMap.class);
+
+        when(resource.getPath()).thenReturn(SITES_PAGE_PATH);
+        ResourceResolver spyResolver = spy(resolver);
+        when(resource.getResourceResolver()).thenReturn(spyResolver);
+        Resource mockPWAResource = mock(Resource.class);
+        when(spyResolver.getResource("/content/mysite/" + JcrConstants.JCR_CONTENT)).thenReturn(mockPWAResource);
+
+        mvp.put("pwaEnabled", true);
+        mvp.put("startURL", SITES_PAGE_PATH + ".html");
+        when(mockPWAResource.getValueMap()).thenReturn(mvp);
+    }
+
+    @Test
+    public void testPWAReturnsManifestPath() {
+        pwa = resource.adaptTo(PWA.class);
+        assertEquals(SITES_PAGE_PATH + "/manifest.webmanifest", pwa.getManifestPath());
+    }
+
+    @Test
+    public void testPWAReturnsServiceWorkerPath() {
+        pwa = resource.adaptTo(PWA.class);
+        assertEquals("/mysitesw.js", pwa.getServiceWorkerPath());
+    }
+
+    @Test
+    public void testPWAReturnsFalseIfPWAOptionIsNotEnabled() {
+        mvp.remove("pwaEnabled");
+        pwa = resource.adaptTo(PWA.class);
+        assertFalse(pwa.isEnabled());
+    }
+
+    @Test
+    public void testPWAReturnsTrueIfPWAOptionIsEnabled() {
+        pwa = resource.adaptTo(PWA.class);
+        assertTrue(pwa.isEnabled());
+    }
+
+    @Test
+    public void testPWAReturnsIconPath() {
+        mvp.put("pwaIcon", "/content/dam/foo.png");
```

Comment:
Please use the defined property names constants when referencing them in tests too! Also at L71, L102, L109, L116, L123, and L130